sound/soc/intel/boards/sof_es8336.c

```diff
@@ -690,13 +763,29 @@ static int sof_es8336_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static const struct platform_device_id board_ids[] = {
+	{
+		.name = "adl_es83x6_hdmi_ssp",
+		.driver_data = (kernel_ulong_t)(SOF_ES8336_SSP_CODEC(1) |
```

Comment:
the adl prefix adds zero information. You need some sort of encoding like

sof_es83x6_c1_h02 /* codec on SSP1, HDMI capture on SSP0 and SSP2 */